ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +52,410 @@ public override string Name
+                        var tokenOld = getTokenFromContainer(key);
+                        var tokenNew = pair.Last;
+
+                        if (tokenOld.Type == JTokenType.Object && tokenNew.Type == JTokenType.Object)
+                        {
+                            deepMergeInto((JObject)tokenOld, (JObject)tokenNew);
+                            addTokenToContainer(key, tokenOld);
+                        }
+                        else if (tokenOld.Type == JTokenType.Array)
+                        {
+                            (tokenOld as JArray).Merge(tokenNew);
+                            addTokenToContainer(key, tokenOld);
+                        }
+                        else if (tokenNew.Type == JTokenType.Array)
+                        {
+                            (tokenNew as JArray).Merge(tokenOld);
+                            addTokenToContainer(key, tokenNew);
+                        }
+                        else if (tokenOld.Type == JTokenType.Null)
+                        {
+                            addTokenToContainer(key, tokenNew);
+                        }
+                    }
+                    else
+                    {
+                        result.Add(new JArray
+                        {
+                            _invalidKey,
+                            pair.First,
+                        });
+                    }
+                }
+                else
+                {
+                    result.Add(new JArray
+                    {
+                        _invalidPair,
+                        keyValue,
+                    });
+                }
+            }
+
+            callback.Invoke(result);
+        }
+
+        /// <summary>
+        /// Clears the <see cref="ApplicationDataContainer"/>.
+        /// When done invokes <see cref="ICallback"/>.
+        /// </summary>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void clear(ICallback callback)
+        {
+            Debug.Assert(callback != null);
+
+            _dataContainer.Values.Clear();
+            _typeContainer.Values.Clear();
+
+            callback.Invoke();
+        }
+
+        /// <summary>
+        /// Returns a <see cref="JArray"/> with all keys from the 
+        /// through <see cref="ICallback"/>.
+        /// </summary>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void getAllKeys(ICallback callback)
+        {
+            Debug.Assert(callback != null);
+
+            callback.Invoke(new JArray(_dataContainer.Values.Keys));
+        }
+
+        /// <summary>
+        /// Adds <see cref="JToken"/> to <see cref="ApplicationDataContainer"/> with the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="token">The token.</param>
+        private void addTokenToContainer(string key, JToken token)
```

Comment:
Can we keep the C# conventions of capitalizing the first letter capitalized for these methods?  Obviously only for the non-React methods.